### DISCUSSION TITLE
Creating a 301 redirect from a full page component

### DISCUSSION THREAD

**Discussion by trsdesign** [livewire/livewire#8745] [7. Bugs (archived)]
### Livewire version

v3.4.10

### Laravel version

v10.48.5

### Which PHP version are you using?

PHP 8.1

### Steps To Reproduce

Create a full page component, issue a redirect in the mount method with a 301 status:

```
class Test extends Component
{
    public function mount()
    {
        return redirect('/', 301);
    }

    public function render()
    {
        return view('livewire.test');
    }
}
```

The redirect is respected, but that status seems to be ignored and it goes through as a 302.

This issue comes from the Livewire Redirector, where the status doesn't appear to be used.

**Livewire\Features\SupportRedirects\Redirector** 

```
class Redirector extends BaseRedirector
{
    protected $component;

    public function to($path, $status = 302, $headers = [], $secure = null)
    {
        $this->component->redirect($this->generator->to($path, [], $secure));

        return $this;
    }
```

### Repository or Wirebox URL

https://417md.wirebox.app:8443/

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by studio-creativo** [comment]
I’m having the same problem. News?

**Reply by lrljoe** [reply]
[https://livewire.laravel.com/docs/redirecting](https://livewire.laravel.com/docs/redirecting)

```
        $this->redirect('/posts'); 
```

What isn't working?  You're going to get a 302 redirect from Laravel regardless, unless you use the Route redirect method:

```
Route::redirect('/here', '/there', 301);
```

**Comment by nox-4O4** [comment]
You can create a 301 redirect by thowing a `HttpResponseException` from your `mount()` function, e.g. using the `abort()` helper:
```php
public function mount() {
    abort(new \Illuminate\Http\RedirectResponse('/', 301));
}
```
